plugin/build_baron.go

```diff
@@ -178,6 +178,29 @@ func BbGetConfig(settings *evergreen.Settings) map[string]evergreen.BuildBaronPr
 }
 
 func BbGetProject(settings *evergreen.Settings, projectId string) (evergreen.BuildBaronProject, bool) {
+	flags, err := evergreen.GetServiceFlags()
+	if err != nil {
+		return evergreen.BuildBaronProject{}, false
+	}
+	if flags.PluginAdminPageDisabled {
+		lastGoodVersion, err := model.FindVersionByLastKnownGoodConfig(projectId, -1)
```

Comment:
Similar to the last PR, there are some places where we call BbGetProject where we can pass in the version ID from the task, and then we only need to call this if that's empty.